src/software/thunderscope/gl/layers/gl_world_layer.py

```diff
@@ -237,6 +239,10 @@ def refresh_graphics(self) -> None:
 
         self.cached_world = self.world_buffer.get(block=False)
 
+        # if not receiving worlds, just render an empty field
+        if is_field_message_empty(self.cached_world.field):
+            self.cached_world = DEFAULT_EMPTY_FIELD_WORLD
```

Comment:
nit: just for extra clarity
```suggestion
        self.cached_world = self.world_buffer.get(block=False, return_cached=True)

        # if not receiving worlds, just render an empty field
        if is_field_message_empty(self.cached_world.field):
            self.cached_world = DEFAULT_EMPTY_FIELD_WORLD
```